
<!DOCTYPE html>
<html lang='en'>
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
    <title>Tree - rpms/grub2 - src.fedoraproject.org</title>
        <link rel="shortcut icon" type="image/vnd.microsoft.icon"
        href="/theme/static/favicon.ico?version=5.10.0"/>
    <link href="/theme/static/fedora-bootstrap-1.3.0/fedora-bootstrap.min.css?version=5.10.0"
        type="text/css" rel="stylesheet" />
    <link href="/theme/static/fonts/fonts.css?version=5.10.0"
        rel="stylesheet" type="text/css" />
    <link href="/theme/static/fonts/hack_fonts/css/hack-extended.min.css?version=5.10.0"
        type="text/css" rel="stylesheet" />
    <link href="/theme/static/theme.css?version=5.10.0"
        type="text/css" rel="stylesheet" />

    <link type="text/css" rel="stylesheet" nonce="CPZp8guieK9V6ID7n1XWMB6c1" href="/static/vendor/font-awesome/font-awesome.css?version=5.10.0"/>
    <link type="text/css" rel="stylesheet" nonce="CPZp8guieK9V6ID7n1XWMB6c1" href="/static/pagure.css?version=5.10.0"/>
<link nonce="CPZp8guieK9V6ID7n1XWMB6c1" rel="stylesheet" href="/static/vendor/highlight.js/styles/github.css?version=5.10.0"/>
<link nonce="CPZp8guieK9V6ID7n1XWMB6c1" rel="stylesheet" href="/static/vendor/highlightjs-line-numbers/highlightjs-line-numbers.min.css?version=5.10.0"/>
<style nonce="CPZp8guieK9V6ID7n1XWMB6c1">
  .hljs {
    background: #fff;
  }
</style>
  </head>
  <body id="home">
    
    <!-- start masthead -->
    <nav class="navbar navbar-light masthead p-0 navbar-expand">
      <div class="container">
        <a href="/" class="navbar-brand">
        <img height=40px src="/theme/static/pagure-logo.png?version=5.10.0"
             alt="pagure Logo" id="pagureLogo"/>
        </a>
        <ul class="navbar-nav ml-auto">
          <li class="nav-item">
            <a class="btn btn-primary" href="/login/?next=https://src.fedoraproject.org/rpms/grub2/blob/master/f/0129-Add-auto-hide-menu-support.patch">Log In</a>
          </li>
        </ul>
      </div>
    </nav>
    <!-- close masthead-->

    <div class="bodycontent">


<div class="bg-light border border-bottom pt-3">
  <div class="container">
    <div class="row mb-3">
      <div class="col-6">
        <div class="row">
        <div class="col-auto pr-0">
            <h3>
<i class="fa fa-archive text-muted"></i></h3>
        </div>
        <div class="col-auto pl-2">
            <h3 class="mb-0">
<a href="/projects/rpms/%2A">rpms</a>&nbsp;/&nbsp;<a href="/rpms/grub2"><strong>grub2</strong></a>
            </h3>
        </div>
        </div>
      </div>
      <div class="col-6 text-right">
        <div class="btn-group">
        <div class="btn-group">
        <a href="#"
            class="btn btn-sm dropdown-toggle btn-outline-primary"
            data-toggle="dropdown" id="watch-button">
          <i class="fa fa-clone fa-fw"></i>
          <span>Clone</span>
        </a>
        <div class="dropdown-menu dropdown-menu-right">
          <div class="m-3" id="source-dropdown" class="pointer">
            <div>
              <h5><strong>Source Code</strong></h5>

              <div class="form-group">
                <div class="input-group input-group-sm">
                  <div class="input-group-prepend"><span class="input-group-text">GIT</span></div>
                  <input class="form-control bg-white select-on-focus" type="text" value="https://src.fedoraproject.org/rpms/grub2.git" readonly>
                </div>
              </div>
            </div>
          </div>
        </div>

      </div>
    </div>
  </div>
</div>

<ul class="nav nav-tabs nav-small border-bottom-0">
  <li class="nav-item mr-2 text-dark">
    <a class="nav-link active" href="/rpms/grub2">
        <i class="fa fa-code fa-fw text-muted"></i>
        <span class="d-none d-md-inline">Source</span>
    </a>
  </li>

    <li class="nav-item mr-2 text-dark">
        <a class="nav-link" href="https://bugzilla.redhat.com/buglist.cgi?bug_status=NEW&bug_status=ASSIGNED&classification=Fedora&product=Fedora&product=Fedora EPEL&component=grub2">
          <i class="fa fa-fw text-muted fa-exclamation-circle"></i>
          <span class="d-none d-md-inline">Issues&nbsp;</span>
          <span class="fa fa-external-link"></span>
        </a>
    </li>



    <li class="nav-item mr-2 text-dark">
      <a class="nav-link" href="/rpms/grub2/stats">
          <i class="fa fa-line-chart fa-fw text-muted"></i>
          <span class="d-none d-md-inline">Stats</span>
      </a>
    </li>


</ul>
  </div>
</div>

<div class="container pt-5 repo-body-container">
  <div class="row">
    <div class="col-2">
<nav class="nav nav-tabs nav-sidetabs flex-column">
  <a class=
      "nav-link nowrap
"
      href="/rpms/grub2">
      <i class="fa fa-home text-muted fa-fw"></i>&nbsp;<span class="d-none d-md-inline">Overview</span>
  </a>
  <a class=
    "nav-link nowrap
 active"
    href="/rpms/grub2/tree/master">
    <i class="fa fa-file-code-o text-muted fa-fw"></i>&nbsp;Files
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/grub2/commits/master">
    <i class="fa fa-list-alt text-muted fa-fw" data-glyph="spreadsheet"></i>&nbsp;Commits
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/grub2/branches?branchname=master">
    <i class="fa fa-random text-muted fa-fw"></i>&nbsp;Branches
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/grub2/forks">
    <i class="fa fa-code-fork text-muted fa-fw"></i>&nbsp;Forks
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/grub2/releases">
    <i class="fa fa-tags text-muted fa-fw"></i>&nbsp;Releases
  </a>

  <div class="col-xs-2 line-height-1"></div>
  <h6>Monitoring status:</h6>
  <div class="btn-group">
    <button title="Monitoring status" class="btn btn-sm btn-outline-primary disabled"
        id="monitoring-button">
      <i id="monitoring-icon" class="fa fa-fw fa-eye"></i>
      <span id="monitoring-label" class="fa fa-circle-o-notch fa-spin fa-1x fa-fw"></span>
    </button>
  </div>


  <div class="pt-3">
    <div class="col-xs-2 line-height-1">
    <h6>Bugzilla Assignee:</h6>
      <dl>
        <dt>Fedora: </dt>
        <dd id="fedora_assignee_txt">
        pjones
        </dd>
        <dt>EPEL: </dt>
        <dd id="epel_assignee_txt">
        pjones
        </dd>
      </dl>
    </div>
  </div>

  <div class="modal fade" id="modal_assignee" tabindex="-1"
          role="dialog" aria-labelledby="Bugzilla assignee" aria-hidden="true">
    <div class="modal-dialog" role="document">
      <div class="modal-content">
        <div class="modal-header">
          <h4 class="modal-title">Bugzilla Assignee</h4>
          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
            <span class="sr-only">Close</span>
          </button>
        </div>
        <form id="change_assignees">
          <div class="modal-body">
            <label for="fedora_assignee">Fedora</label>
            <input title="Default assignee for Fedora in bugzilla - Empty input resets to default"
              class="form-control" name="fedora_assignee" id="fedora_assignee" value="pjones"/>
            <label for="epel_assignee">EPEL</label>
            <input title="Default assignee for EPEL in bugzilla (if applicable) - Empty input resets to default"
              class="form-control" name="epel_assignee" id="epel_assignee" value="pjones" />
            <p class="pt-2">
              These two fields allow to specify a different default assignee for ticket opened against
              this package in bugzilla. Note: The EPEL field is
              always displayed for packages in the 'rpms' namespace regardless of whether it
              is used in bugzilla or not.            </p>
          </div>
          <div class="modal-footer">
            <button class="btn btn-secondary" type="button" title="Update bugzilla overrides" id="reset_assignees">
               Reset to defaults
            </button>
            <button class="btn btn-primary" type="submit" title="Update bugzilla overrides" id="update_assignees">
               Update
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>

  <script type="text/javascript" nonce="CPZp8guieK9V6ID7n1XWMB6c1">
    window.addEventListener('load', function() {
      set_up_monitoring = function(status){
        var _label = "Disabled"
        if (status === "monitoring") {
          _label = "Monitoring";
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else if (status === "monitoring-with-scratch") {
          _label = "Scratch builds"
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else {
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye-slash")
        }

        $("#monitoring-label").text(_label);
        $("#monitoring-label").removeClass("fa fa-circle-o-notch fa-spin fa-1x fa-fw");
      }

      $.ajax({
        url: "/_dg/anitya/rpms/grub2",
        type: 'GET',
        dataType: 'json',
        success: function(res) {
          console.log(res);
          set_up_monitoring(res.monitoring)
        }
      });


      $("#reset_assignees").on('click', function(){
        $('#fedora_assignee').val('');
        $('#epel_assignee').val('');
        $("#change_assignees").submit();
        return false;
      });

      $("#change_assignees").on('submit',  function(){
        $('html').css('cursor', 'progress');
        $('#reset_assignees').attr('disabled', true);
        $('#update_assignees').attr('disabled', true);
        $('#update_assignees').text('Updating...');
        $.ajax({
          url: "/_dg/bzoverrides/rpms/grub2",
          type: 'POST',
          dataType: 'json',
          data: {
             'epel_assignee': $('#epel_assignee').val(),
             'fedora_assignee': $('#fedora_assignee').val()
          },
          success: function(res) {
            $("#fedora_assignee_txt").text(res.fedora_assignee);
            $("#epel_assignee_txt").text(res.epel_assignee);
            $('#modal_assignee').modal('hide');
            $('#reset_assignees').attr('disabled', false);
            $('#update_assignees').attr('disabled', false);
            $('#update_assignees').text('Update');
            $('html').css('cursor', 'default');
            console.log("Successfully changed the bugzilla assignees");
            return false;
          },
          error: function(res) {
            var msg = '';
            if(res.responseJSON.errors){
              msg = ': ' + res.responseJSON.errors.join(', ');
            }
            alert("Unable to update the bugzilla assignee(s)" + msg);
            $('html').css('cursor', 'default');
            $('#reset_assignees').attr('disabled', false);
            $('#update_assignees').attr('disabled', false);
            $('#update_assignees').text('Update');
            return false;
          }
        })
        return false;
      });


    });
  </script>

</nav>    </div>
    <div class="col-10">
  <div class="row mb-1">
    <div class="col-sm-6">
    <h3>
      Files
    </h3>
    </div>

    <div class="col-sm-6">
      <div class="float-right">
          <div class="btn-group">
            <a href="#" class="btn btn-outline-light border-secondary text-dark btn-sm dropdown-toggle"
                    data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                    <span class="fa fa-random fa-fw"></span> Branch: <span class="font-weight-bold">master</span>
          </a>
            <div class="dropdown-menu dropdown-menu-right">
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/el5">
                      <span class="fa fa-random fa-fw"></span> <span class="">el5</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/el6">
                      <span class="fa fa-random fa-fw"></span> <span class="">el6</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f10">
                      <span class="fa fa-random fa-fw"></span> <span class="">f10</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f11">
                      <span class="fa fa-random fa-fw"></span> <span class="">f11</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f12">
                      <span class="fa fa-random fa-fw"></span> <span class="">f12</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f13">
                      <span class="fa fa-random fa-fw"></span> <span class="">f13</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f14">
                      <span class="fa fa-random fa-fw"></span> <span class="">f14</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f15">
                      <span class="fa fa-random fa-fw"></span> <span class="">f15</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f16">
                      <span class="fa fa-random fa-fw"></span> <span class="">f16</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f17">
                      <span class="fa fa-random fa-fw"></span> <span class="">f17</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f18">
                      <span class="fa fa-random fa-fw"></span> <span class="">f18</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f19">
                      <span class="fa fa-random fa-fw"></span> <span class="">f19</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f20">
                      <span class="fa fa-random fa-fw"></span> <span class="">f20</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f21">
                      <span class="fa fa-random fa-fw"></span> <span class="">f21</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f22">
                      <span class="fa fa-random fa-fw"></span> <span class="">f22</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f23">
                      <span class="fa fa-random fa-fw"></span> <span class="">f23</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f24">
                      <span class="fa fa-random fa-fw"></span> <span class="">f24</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f25">
                      <span class="fa fa-random fa-fw"></span> <span class="">f25</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f25-bls">
                      <span class="fa fa-random fa-fw"></span> <span class="">f25-bls</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f25-bls3">
                      <span class="fa fa-random fa-fw"></span> <span class="">f25-bls3</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f26">
                      <span class="fa fa-random fa-fw"></span> <span class="">f26</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f27">
                      <span class="fa fa-random fa-fw"></span> <span class="">f27</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f28">
                      <span class="fa fa-random fa-fw"></span> <span class="">f28</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f29">
                      <span class="fa fa-random fa-fw"></span> <span class="">f29</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f30">
                      <span class="fa fa-random fa-fw"></span> <span class="">f30</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f31">
                      <span class="fa fa-random fa-fw"></span> <span class="">f31</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f32">
                      <span class="fa fa-random fa-fw"></span> <span class="">f32</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/f9">
                      <span class="fa fa-random fa-fw"></span> <span class="">f9</span>
                  </a>
                  <a class="dropdown-item pl-1 active" href="/rpms/grub2/tree/master">
                      <span class="fa fa-random fa-fw"></span> <span class="font-weight-bold">master</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/grub2/tree/splitarch4">
                      <span class="fa fa-random fa-fw"></span> <span class="">splitarch4</span>
                  </a>
            </div>
          </div>
    </div>
  </div>
  </div>
    <div class="card mb-3">
      <div class="card-header">
        <ol class="breadcrumb p-0 bg-transparent mb-0">
          <li class="breadcrumb-item">
            <a href="/rpms/grub2/tree/master">
              <span class="fa fa-random">
              </span>&nbsp; master
            </a>
          </li>
          <li class="active breadcrumb-item">
            <span class="fa fa-file" data-glyph="">
            </span>&nbsp; 0129-Add-auto-hide-menu-support.patch
          </li>
        </ol>
      </div>

  <div class="card-body p-0">
            <div class="bg-light border text-right pr-2">
                <form class="btn btn-sm" method="POST" name="fork_project"
                    action="/fork_edit/rpms/grub2/edit/master/f/0129-Add-auto-hide-menu-support.patch">
                    <button class="btn btn-sm btn-secondary fork_project_btn">
                            Fork and Edit
                    </button>
                    
                </form>

                <a class="btn btn-secondary btn-sm" href="/rpms/grub2/blob/master/f/0129-Add-auto-hide-menu-support.patch" title="View as blob">Blob</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/grub2/blame/0129-Add-auto-hide-menu-support.patch?identifier=master" title="View git blame">Blame</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/grub2/history/0129-Add-auto-hide-menu-support.patch?identifier=master" title="View git log for this file">History</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/grub2/raw/master/f/0129-Add-auto-hide-menu-support.patch" title="View as raw">Raw</a>
            </div>

        <pre class="syntaxhighlightblock"><code class="lang-diff">From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Hans de Goede &lt;hdegoede@redhat.com&gt;
Date: Wed, 6 Jun 2018 08:44:11 +0200
Subject: [PATCH] Add auto-hide menu support

On single-os systems we do not want to show the menu, unless something
went wrong with the previous boot, in which case the user may need the
menu to debug/fix the problem.

This commit adds a new grub.d/00_menu_auto_hide file which emits a
config snippet implementing this. I&#39;ve chosen to do this in a separate
grub.d file because chances of this going upstream are small and this way
it will be easier to rebase.

Since auto-hiding the menu requires detecting the previous boot was ok,
we get fastboot support (where we don&#39;t check for a key at all) for free
so this commit also adds support for this.

The new config-file code uses the following variables:

menu_auto_hide     Set this to &#34;1&#34; to activate the new auto-hide feature
                   Set this to &#34;2&#34; to auto-hide the menu even when multiple
                   operating systems are installed. Note the menu will still
                   auto show after booting an other os as that won&#39;t set
                   boot_success.
menu_show_once     Set this to &#34;1&#34; to force showing the menu once.
boot_success       The OS sets this to &#34;1&#34; to indicate a successful boot.
boot_indeterminate The OS increments this integer when rebooting after e.g.
                   installing updates or a selinux relabel.
fastboot           If set to &#34;1&#34; and the conditions for auto-hiding the menu
                   are met, the menu is not shown and all checks for keypresses
                   are skipped, booting the default immediately.

30_os-prober.in changes somewhat inspired by:
https://git.launchpad.net/~ubuntu-core-dev/grub/+git/ubuntu/tree/debian/patches/quick_boot.patch

Signed-off-by: Hans de Goede &lt;hdegoede@redhat.com&gt;
---
Changes in v2:
-Drop shutdown_success tests, there is no meaningful way for systemd to set
 this flag (by the time it knows all filesystems are unmounted or read-only
-Drop fwsetup_once support, systemd already supports booting directly into
 the fwsetup by doing &#34;systemctl reboot --firmware&#34;
---
 Makefile.util.def                |  6 +++++
 util/grub.d/01_menu_auto_hide.in | 48 ++++++++++++++++++++++++++++++++++++++++
 util/grub.d/30_os-prober.in      | 18 +++++++++++++++
 3 files changed, 72 insertions(+)
 create mode 100644 util/grub.d/01_menu_auto_hide.in

diff --git a/Makefile.util.def b/Makefile.util.def
index c13ca685ce1..026b458bb85 100644
--- a/Makefile.util.def
+++ b/Makefile.util.def
@@ -457,6 +457,12 @@ script = {
   installdir = grubconf;
 };
 
+script = {
+  name = &#39;01_menu_auto_hide&#39;;
+  common = util/grub.d/01_menu_auto_hide.in;
+  installdir = grubconf;
+};
+
 script = {
   name = &#39;01_users&#39;;
   common = util/grub.d/01_users.in;
diff --git a/util/grub.d/01_menu_auto_hide.in b/util/grub.d/01_menu_auto_hide.in
new file mode 100644
index 00000000000..ad175870a54
--- /dev/null
+++ b/util/grub.d/01_menu_auto_hide.in
@@ -0,0 +1,48 @@
+#! /bin/sh
+
+# Disable / skip generating menu-auto-hide config parts on serial terminals
+for x in ${GRUB_TERMINAL_INPUT} ${GRUB_TERMINAL_OUTPUT}; do
+  case &#34;$x&#34; in
+    serial*)
+      exit 0
+      ;;
+  esac
+done
+
+cat &lt;&lt; EOF
+if [ &#34;\${boot_success}&#34; = &#34;1&#34; -o &#34;\${boot_indeterminate}&#34; = &#34;1&#34; ]; then
+  set last_boot_ok=1
+else
+  set last_boot_ok=0
+fi
+
+# Reset boot_indeterminate after a successful boot
+if [ &#34;\${boot_success}&#34; = &#34;1&#34; ] ; then
+  set boot_indeterminate=0
+# Avoid boot_indeterminate causing the menu to be hidden more then once
+elif [ &#34;\${boot_indeterminate}&#34; = &#34;1&#34; ]; then
+  set boot_indeterminate=2
+fi
+set boot_success=0
+save_env boot_success boot_indeterminate
+
+if [ x\$feature_timeout_style = xy ] ; then
+  if [ &#34;\${menu_show_once}&#34; ]; then
+    unset menu_show_once
+    save_env menu_show_once
+    set timeout_style=menu
+    set timeout=60
+  elif [ &#34;\${menu_auto_hide}&#34; -a &#34;\${last_boot_ok}&#34; = &#34;1&#34; ]; then
+    set orig_timeout_style=\${timeout_style}
+    set orig_timeout=\${timeout}
+    if [ &#34;\${fastboot}&#34; = &#34;1&#34; ]; then
+      # timeout_style=menu + timeout=0 avoids the countdown code keypress check
+      set timeout_style=menu
+      set timeout=0
+    else
+      set timeout_style=hidden
+      set timeout=1
+    fi
+  fi
+fi
+EOF
diff --git a/util/grub.d/30_os-prober.in b/util/grub.d/30_os-prober.in
index 13a3a6bc752..ab634393a31 100644
--- a/util/grub.d/30_os-prober.in
+++ b/util/grub.d/30_os-prober.in
@@ -42,6 +42,7 @@ if [ -z &#34;${OSPROBED}&#34; ] ; then
 fi
 
 osx_entry() {
+    found_other_os=1
     # TRANSLATORS: it refers on the OS residing on device %s
     onstr=&#34;$(gettext_printf &#34;(on %s)&#34; &#34;${DEVICE}&#34;)&#34;
     hints=&#34;&#34;
@@ -102,6 +103,7 @@ for OS in ${OSPROBED} ; do
 
   case ${BOOT} in
     chain)
+      found_other_os=1
 
 	  onstr=&#34;$(gettext_printf &#34;(on %s)&#34; &#34;${DEVICE}&#34;)&#34;
       cat &lt;&lt; EOF
@@ -132,6 +134,7 @@ EOF
 EOF
     ;;
     efi)
+      found_other_os=1
 
 	EFIPATH=${DEVICE#*@}
 	DEVICE=${DEVICE%@*}
@@ -176,6 +179,7 @@ EOF
 	  LINITRD=&#34;${LINITRD#/boot}&#34;
 	fi
 
+        found_other_os=1
 	onstr=&#34;$(gettext_printf &#34;(on %s)&#34; &#34;${DEVICE}&#34;)&#34;
 	recovery_params=&#34;$(echo &#34;${LPARAMS}&#34; | grep single)&#34; || true
 	counter=1
@@ -249,6 +253,7 @@ EOF
       done
     ;;
     hurd)
+      found_other_os=1
       onstr=&#34;$(gettext_printf &#34;(on %s)&#34; &#34;${DEVICE}&#34;)&#34;
       cat &lt;&lt; EOF
 menuentry &#39;$(echo &#34;${LONGNAME} $onstr&#34; | grub_quote)&#39; --class hurd --class gnu --class os \$menuentry_id_option &#39;osprober-gnuhurd-/boot/gnumach.gz-false-$(grub_get_device_id &#34;${DEVICE}&#34;)&#39; {
@@ -275,6 +280,7 @@ EOF
 EOF
     ;;
     minix)
+      found_other_os=1
 	  cat &lt;&lt; EOF
 menuentry &#34;${LONGNAME} (on ${DEVICE}, Multiboot)&#34; {
 EOF
@@ -306,3 +312,15 @@ EOF
       esac
   esac
 done
+
+# We override the results of the menu_auto_hide code here, this is a bit ugly,
+# but grub-mkconfig writes out the file linearly, so this is the only way
+if [ &#34;${found_other_os}&#34; = &#34;1&#34; ]; then
+  cat &lt;&lt; EOF
+# Other OS found, undo autohiding of menu unless menu_auto_hide=2
+if [ &#34;\${orig_timeout_style}&#34; -a &#34;\${menu_auto_hide}&#34; != &#34;2&#34; ]; then
+  set timeout_style=\${orig_timeout_style}
+  set timeout=\${orig_timeout}
+fi
+EOF
+fi
</code></pre>
  </div>
 </div> <!-- end .card-->

</div>
</div>
</div>
    </div>

        <div class="footer pt-4 text-white">
        <div class="container">
            <div class="d-flex align-items-center">
                <div>
                    <div>Powered by <a href="https://pagure.io/pagure" class="notblue">Pagure</a> 5.10.0</div>
                    <div>
                        <a href="https://docs.pagure.org/pagure/usage/index.html" class="notblue">Documentation</a> &bull;
                        <a href="https://pagure.io/pagure/new_issue" class="notblue">File an Issue</a> &bull;
                        <a href="/about">About this Instance</a> &bull;
                        <a href="/ssh_info" class="notblue">SSH Hostkey/Fingerprint</a>
                    </div>
                </div>
                <div class="ml-auto text-right">
                    <div>&copy; Red Hat, Inc. and others.</div>
                </div>
            </div>
        </div>
    </div>


    <script type="text/javascript" nonce="CPZp8guieK9V6ID7n1XWMB6c1" src="/static/vendor/jquery/jquery.min.js?version=5.10.0"></script>

    <script src="/static/vendor/bootstrap/bootstrap.bundle.min.js?version=5.10.0"></script>

    <script type="text/javascript" nonce="CPZp8guieK9V6ID7n1XWMB6c1">
      $('[data-toggle="tooltip"]').tooltip({placement : 'bottom'});
      $(".cancel_btn").click(function() {
        history.back();
      });
    </script>

<script type="text/javascript"  nonce="CPZp8guieK9V6ID7n1XWMB6c1" src="/static/vendor/lazyload/lazyload.min.js?version=5.10.0"></script>

<script type="text/javascript" nonce="CPZp8guieK9V6ID7n1XWMB6c1">
window.addEventListener("load", function(event) {
    lazyload();
});
</script>

<script type="text/javascript" nonce="CPZp8guieK9V6ID7n1XWMB6c1">
$("#giturl-toggle").on('click', function(event){
  event.stopPropagation();
  $("#giturl-more").toggle();
  $("#giturl-toggle").hide();
})

$(".fork_project_btn").click(function() {
  $('#fork_project').submit();
});

$(".select-on-focus").on("focus", function() {
  $(this).select();
});

</script>

<script type="text/javascript" nonce="CPZp8guieK9V6ID7n1XWMB6c1" src="/static/vendor/highlight.js/highlight.pack.js?version=5.10.0"></script>
<script type="text/javascript" nonce="CPZp8guieK9V6ID7n1XWMB6c1" src="/static/vendor/highlightjs-line-numbers/highlightjs-line-numbers.min.js?version=5.10.0"></script>
<script type="text/javascript" nonce="CPZp8guieK9V6ID7n1XWMB6c1" src="/static/vendor/highlight.js/spec.js?version=5.10.0"></script>

<script type="text/javascript" nonce="CPZp8guieK9V6ID7n1XWMB6c1">
  $(document).ready(function() {
  $('.fork_project_btn').click($("[name=fork_project]").submit);

  $('pre.syntaxhighlightblock code').each(function(i, block) {
    hljs.highlightBlock(block);
    hljs.lineNumbersBlock(block);
  });

  var cls = "highlighted-line";
  var lines = location.hash.substr(2).split('-').map(function (x) { return parseInt(x, 10) });
  if (! isNaN(lines[0]))
  {
    for (var i = lines[lines.length - 1]; i >= lines[0]; i--) {
      $('#_' + i).parent().parent().addClass(cls);
    }
    setTimeout(function(){
      $("#_" + lines[0]).get(0).scrollIntoView({behavior: "instant", block: "start", inline: "nearest"});
    }, 50);
  }
});
</script>

<script type="text/javascript" nonce="CPZp8guieK9V6ID7n1XWMB6c1">

  function updateHighlight() {
    var cls = "highlighted-line";
    $('.' + cls).removeClass(cls)
    if (location.hash !== '') {
      var lines = location.hash.substr(2).split('-').map(function (x) { return parseInt(x, 10) });
      for (var i = lines[lines.length - 1]; i >= lines[0]; i--) {
        $('[data-line-number=' + i + ']').closest('tr').addClass(cls);
      }
      return lines;
    }
    return [];
  }
  $(window).on('hashchange', updateHighlight);
  var selected = [];
  $("[data-line-number]").click(function (ev) {
    var line = $(this).attr('data-line-number');
    if (ev.shiftKey) {
      selected = selected.slice(-1).concat(line);
    } else {
      selected = [line];
    }

    var hash = '_' + selected[0];
    if (selected.length === 2) {
      hash = '_' + Math.min(selected[0], selected[1]) + '-' + Math.max(selected[0], selected[1]);
    }
    window.location.hash = hash;
    return false;
  });

</script>


</body>
</html>